libbpf-rs/src/program.rs

```diff
@@ -624,6 +624,30 @@ impl<'obj> Program<'obj> {
         Ok(prog_info.id)
     }
 
+    /// Returns program fd by pin
+    pub fn get_fd_by_pin<P: AsRef<Path>>(path: P) -> Result<OwnedFd> {
+        let path_c = util::path_to_cstring(path)?;
+        let path_ptr = path_c.as_ptr();
+        let fd = util::parse_ret_i32(unsafe { libbpf_sys::bpf_obj_get(path_ptr) })?;
```

Comment:
Can you please add context to fallible calls (not just this one):
```suggestion
        let fd = util::parse_ret_i32(unsafe { libbpf_sys::bpf_obj_get(path_ptr) })
            .with_context(|| format!("failed to retrieve BPF object from path `{}`", path.as_ref().display()))?;
```
(may not compile as-is but you get the idea)

Reply:
Added.